internal/remoteconfig/remoteconfig.go

```diff
@@ -225,13 +232,13 @@ func (c *Client) applyUpdate(pbUpdate *clientGetConfigsResponse) error {
 		updatedProducts[p] = struct{}{}
 	}
 
-	// Performs the callbacks registered for all updated products and update the application status in the repository
-	// (RCTE2)
-	for p := range updatedProducts {
-		for _, fn := range c.callbacks[p] {
-			for path, status := range fn(productUpdates[p]) {
-				c.repository.UpdateApplyStatus(path, status)
-			}
+	if len(updatedProducts) == 0 {
+		return nil
+	}
+	// Performs the callbacks registered and update the application status in the repository (RCTE2)
+	for _, fn := range c.callbacks {
+		for path, status := range fn(productUpdates) {
+			c.repository.UpdateApplyStatus(path, status)
```

Comment:
In the case of several callbacks, listening the same products and files, the last one will enforce its apply status.
WDYT of merging the maps returning by the N calls to `fn(productUpdates)` instead and updating the apply status only once at the end of the callbacks loop?

Reply:
I don't think this case should happen since each config path is product specific and a product is likely to only be handled by a single component (ASM, APM, Profiler, etc...) but added it for safety, see 067085cb20fd64451